test/distributed/fsdp/test_fsdp_dtensor_state_dict.py

```diff
@@ -129,6 +132,50 @@ def test_dtensor_sharded_model_state_dict(self, offload_to_cpu):
                 # check whether device are the same
                 self.assertEqual(v1.to_local().device, v2.local_tensor().device)
 
+    @with_comms
+    @skip_if_lt_x_gpu(2)
+    @parametrize("map_location", ["cpu", "cuda"])
+    def test_dtensor_sharded_model_load_state_dict(self, map_location):
+        model = FSDP(TestDummyModel().cuda())
+        optim = torch.optim.Adam(model.parameters(), lr=0.1)
+
+        FSDP.set_state_dict_type(
+            model,
+            StateDictType.SHARDED_STATE_DICT,
+            state_dict_config=ShardedStateDictConfig(
+                use_dtensor=True,
+                offload_to_cpu=False,
+            ),
+        )
+
+        checkpoint = io.BytesIO()
+        torch.save(model.state_dict(), checkpoint)
+        # Deepcopy to save current state_dict to compare with the loaded state dict below.
+        ref_state_dict = deepcopy(model.state_dict())
+        # Hmmm...Why does ```torch.save(ref_state_dict, checkpoint)``` not work while torch.save(model.state_dict()) work?
+
+        # Update the parameters so model.state_dict() will be different from ref_dtensor_sd.
+        model(model.get_input()).sum().backward()
+        optim.step()
+
+        # # Load ref_state_dict back.
+        checkpoint.seek(0)
+        load_ref_state_dict = torch.load(checkpoint, map_location=map_location)
+        model.load_state_dict(load_ref_state_dict)
+        new_state_dict = model.state_dict()
+
+        # Check whether new_state_dict is the same as ref_state_dict.
+        for ref_state_dict, new_state_dict in zip(
+            ref_state_dict.items(), new_state_dict.items()
+        ):
+            k1, v1 = ref_state_dict
```

Comment:
this is a little confusing, we're calling `ref_state_dict` the (key, value) tuple, but `ref_state_dict` is also the name of the dict. Can we just do:

`for (k1, v1), (k2, v2) in zip(ref_state_dict.items, ...)`